cpp/include/cudf/contiguous_split.hpp

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/table/table.hpp>
+#include <cudf/types.hpp>
+
+#include <memory>
+#include <vector>
+
+namespace cudf {
+
+/**
+ * @addtogroup column_copy
+ * @{
+ * @file
+ * @brief Table APIs for contiguous_split, pack, unpack, and metadadata
+ */
+
+/**
+ * @brief Column data in a serialized format
+ *
+ * @ingroup copy_split
+ *
+ * Contains data from an array of columns in two contiguous buffers: one on host, which contains
+ * table metadata and one on device which contains the table data.
+ */
+struct packed_columns {
+  packed_columns()
+    : metadata_(std::make_unique<std::vector<uint8_t>>()),
+      gpu_data(std::make_unique<rmm::device_buffer>())
+  {
+  }
```

Comment:
```suggestion
  packed_columns() = default;
```

Reply:
There is code depending on the default constructor to instantiate the `unique_ptrs` (since we need to keep these for now). If these were members directly of packed_columns I am with you, but as it is I can't remove.